extension.js

```diff
@@ -3,19 +3,24 @@
 const ExtensionUtils = imports.misc.extensionUtils;
 const Me = ExtensionUtils.getCurrentExtension();
 const Docking = Me.imports.docking;
+const DesktopIconsIntegration = Me.imports.desktopIconsIntegration;
 
 // We declare this with var so it can be accessed by other extensions in
 // GNOME Shell 3.26+ (mozjs52+).
 var dockManager;
+var DesktopIconsUsableArea;
 
 function init() {
     ExtensionUtils.initTranslations('dashtodock');
 }
 
 function enable() {
+    DesktopIconsUsableArea = new DesktopIconsIntegration.DesktopIconsUsableAreaClass();
```

Comment:
Use DockManager for this

Reply:
I don't understand. DockManager?

Reply:
I'm saying to handle the lifecycle of this object in `Docking.DockManager()` (and `Docking.destroy()`).

If you need to expose this as part of the extension, just get it from that class via a property... I'd prefer to have all managed there.

Reply:
Done.